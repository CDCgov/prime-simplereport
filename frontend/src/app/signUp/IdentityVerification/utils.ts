import * as yup from "yup";

const EXPERIAN_ANSWER_KEY = "outWalletAnswer";

export const getAnswerKey = (index: number) =>
  `${EXPERIAN_ANSWER_KEY}${index + 1}`;

export const toOptions = (
  choices: string[]
): { value: string; label: string }[] =>
  choices.map((choice, index) => {
    return { value: String(index + 1), label: choice };
  });

export const initAnswers = (questionSet: Question[]): Nullable<Answers> =>
  questionSet.reduce((answers, _question, index) => {
    answers[getAnswerKey(index)] = null;
    return answers;
  }, {} as Nullable<Answers>);

export const buildSchema = (questionSet: Question[]): yup.SchemaOf<Answers> =>
  yup.object(
    questionSet.reduce((answers, _question, index) => {
      answers[getAnswerKey(index)] = yup
        .string()
        .nullable()
        .required("This field is required");
      return answers;
    }, {} as { [key: string]: any })
  );

// Note: the order of the answers must be preserved
export const answersToArray = (answers: Answers): string[] =>
  Object.keys(answers).map((_, index) => {
    return answers[getAnswerKey(index)];
  });

export const personalDetailsFields = [
  ["firstName", "First name", true, "Legal name"],
  ["middleName", "Middle name", false, null],
  ["lastName", "Last name", true, null],
  ["dateOfBirth", "Date of birth", true, null],
  ["email", "Email", true, "Personal contact information"],
  ["phoneNumber", "Phone number", true, null],
  ["streetAddress1", "Street address 1", true, "Home address"],
  ["streetAddress2", "Street address 2", false, null],
  ["city", "City", true, null],
  ["state", "State", true, null],
  ["zip", "ZIP code", true, null],
  ["poBoxNumber", "PO Box number", false, null],
].reduce((fields, field) => {
  fields[field[0] as keyof IdentityVerificationRequest] = {
    label: field[1] as string,
    required: field[2] as boolean,
    preheader: field[3] as string | null,
  };
  return fields;
}, {} as { [key: string]: { label: string; required: boolean; preheader: string | null } });

export const initPersonalDetails = (): IdentityVerificationRequest => ({
  firstName: "",
  lastName: "",
  dateOfBirth: "",
  email: "",
  phoneNumber: "",
  streetAddress1: "",
  city: "",
  state: "",
  zip: "",
});

export const initPersonalDetailsErrors = (): Record<
  keyof IdentityVerificationRequest,
  string
> => ({
  firstName: "",
  middleName: "",
  lastName: "",
  dateOfBirth: "",
  email: "",
  phoneNumber: "",
  streetAddress1: "",
  streetAddress2: "",
  city: "",
  state: "",
  zip: "",
  poBoxNumber: "",
});

export const personalDetailsSchema: yup.SchemaOf<IdentityVerificationRequest> = yup
  .object()
  .shape({
    firstName: yup.string().required("First name is required"),
    middleName: yup.string().nullable(),
    lastName: yup.string().required("Last name is required"),
    dateOfBirth: yup.string().required("Birth date is required"),
    email: yup.string().email().required("Email is required"),
    phoneNumber: yup.string().required("Phone number is required"),
    streetAddress1: yup.string().required("Street address is required"),
    streetAddress2: yup.string().nullable(),
    city: yup.string().required("City is required"),
    state: yup.string().required("State is required"),
    zip: yup.string().required("ZIP code is required"),
    poBoxNumber: yup.string().nullable(),
  });
