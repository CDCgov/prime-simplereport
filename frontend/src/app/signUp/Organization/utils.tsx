import * as yup from "yup";
import { ReactElement } from "react";

import { TextWithTooltip } from "../../commonComponents/TextWithTooltip";
import { phoneNumberIsValid } from "../../patients/personSchema";
import { liveJurisdictions } from "../../../config/constants";
import Alert from "../../commonComponents/Alert";

import { OrganizationCreateRequest } from "./OrganizationForm";

export const OrganizationTypeEnum: { [key in OrganizationType]: string } = {
  airport: "Airport/Transit station",
  assisted_living: "Assisted living facility",
  camp: "Camp",
  university: "College/University",
  correctional_facility: "Correctional facility",
  employer: "Employer",
  fqhc: "Federally Qualified Health Center (FQHC)",
  government_agency: "Government agency",
  shelter: "Homeless shelter",
  hospice: "Hospice",
  hospital: "Hospital or clinic",
  k12: "K-12 school",
  lab: "Lab",
  nursing_home: "Nursing home",
  other: "Other",
  pharmacy: "Pharmacy",
  primary_care: "Primary care/Mental health outpatient",
  treatment_center: "Substance use disorder treatment center",
  urgent_care: "Urgent care",
};

// This object contains a map of all the fields in the form
// It also contains the input labels and whether the input is required
// The object is iterated over in OrganizationForm.tsx using
// Object.entries() to generate the form
export const organizationFields = [
  [
    "name",
    <TextWithTooltip
      text="Organization name"
      tooltip="Organizations have multiple testing facilities or locations as part of their network."
      position="right"
    />,
    true,
    "",
  ],
  ["state", "Organization state", true, ""],
  ["type", "Organization type", true, ""],
  ["banner", "Organization administrator", false, ""],
  ["firstName", "First name", true, ""],
  ["middleName", "Middle name", false, ""],
  ["lastName", "Last name", true, ""],
  ["email", "Work email", true, "Enter your individual work email address."],
  [
    "workPhoneNumber",
    "Work phone number",
    true,
    "Enter your direct work phone number.",
  ],
].reduce((fields, field) => {
  fields[field[0] as keyof OrganizationCreateRequest] = {
    label: field[1] as string | React.ReactNode,
    required: field[2] as boolean,
    hintText: field[3] as string,
  };
  return fields;
}, {} as { [key: string]: { label: string | React.ReactNode; required: boolean; hintText: string } });

export const initOrg = (): OrganizationCreateRequest => ({
  name: "",
  type: "",
  state: "",
  firstName: "",
  middleName: "",
  lastName: "",
  email: "",
  workPhoneNumber: "",
});

export const initOrgErrors = (): Record<
  keyof OrganizationCreateRequest,
  string
> => ({
  name: "",
  type: "",
  state: "",
  firstName: "",
  middleName: "",
  lastName: "",
  email: "",
  workPhoneNumber: "",
});

const getStateErrorMessage = (param: any) =>
  param.value !== "" ? (
    <>
      SimpleReport isn't available yet in your state. View{" "}
      <a
        href="https://simplereport.gov/using-simplereport/manage-facility-info/find-supported-jurisdictions"
        target="_blank"
        rel="noopener noreferrer"
      >
        supported jurisdictions
      </a>{" "}
      or sign up for our{" "}
      <a
        href="https://simplereport.gov/waitlist"
        target="_blank"
        rel="noopener noreferrer"
      >
        waitlist
      </a>
      .
    </>
  ) : (
    "Organization state is required"
  );

export const organizationSchema: yup.SchemaOf<OrganizationCreateRequest> = yup
  .object()
  .shape({
    name: yup.string().required("Organization name is required"),
    type: yup
      .mixed()
      .oneOf(Object.keys(OrganizationTypeEnum), "Organization type is required")
      .required(),
    state: yup
      .mixed()
      .oneOf(liveJurisdictions, getStateErrorMessage)
      .required(),
    firstName: yup.string().required("First name is required"),
    middleName: yup.string().nullable(),
    lastName: yup.string().required("Last name is required"),
    email: yup
      .string()
      .email("A valid email address is required")
      .required("A valid email address is required"),
    workPhoneNumber: yup
      .mixed()
      .test("", "A valid phone number is required", phoneNumberIsValid)
      .required(),
  });

export const organizationBackendErrors = (error: string): ReactElement => {
  switch (error) {
    // Unusable org name.  Name probably doesn't contain any alphanumeric characters
    case "The organization name is empty.":
    case "The organization name is invalid.":
      return (
        <Alert type="error" title="Invalid organization name">
          The organization name you entered is invalid. Please double check and
          re-enter the organization name.
        </Alert>
      );
    // Duplicate org. Admin user is attempting to resign up but has already completed identity verification.
    case "Duplicate organization with admin user who has completed identity verification.":
      return (
        <Alert type="error" title="Duplicate organization">
          Your organization is already registered with SimpleReport. Check your
          email for instructions on setting up your account.
        </Alert>
      );
    // Duplicate org. Admin user is attempting to resign up and hasn't finished identity verification.
    case "Duplicate organization with admin user that has not completed identity verification.":
      return (
        <Alert type="error" title="Duplicate organization">
          Your organization is already registered with SimpleReport. To begin
          using it, schedule a time to complete our{" "}
          <a href="https://outlook.office365.com/owa/calendar/Helpdesk@HHSGOV.onmicrosoft.com/bookings/s/HOX7KgZruESJksIgC8qVxQ2">
            online identity verification.
          </a>{" "}
        </Alert>
      );
    // Duplicate org. Non-admin user is attempting to reregister the organization.
    case "This organization has already registered with SimpleReport.":
      return (
        <Alert type="error" title="Duplicate organization">
          This organization already has a SimpleReport account. Please contact
          your organization administrator to request access.
        </Alert>
      );
    // Email already exists in SimpleReport (user is potentially already in another org.)
    case "This email address is already associated with a SimpleReport user.":
      return (
        <Alert type="error" title="Email already registered">
          This email address is already registered with SimpleReport. Please
          contact your organization administrator or{" "}
          <a href="mailto:support@simplereport.gov">support@simplereport.gov</a>{" "}
          for help.
        </Alert>
      );
    // Okta internal error.
    case "An unknown error occurred when creating this organization in Okta.":
      return (
        <Alert type="error" title="Unexpected error">
          An unexpected error occurred. Please resubmit this form, or contact{" "}
          <a href="mailto:support@simplereport.gov">support@simplereport.gov</a>{" "}
          if the problem persists.
        </Alert>
      );
    default:
      return (
        <Alert type="error" title="Submission error">
          {error} Please contact your organization administrator or{" "}
          <a href="mailto:support@simplereport.gov">support@simplereport.gov</a>{" "}
          for help.
        </Alert>
      );
  }
};
