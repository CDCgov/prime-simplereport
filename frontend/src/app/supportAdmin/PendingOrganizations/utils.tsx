import * as yup from "yup";

import { phoneNumberIsValid } from "../../patients/personSchema";

export interface PendingOrganizationFormValues {
  name: string;
  adminFirstName: string | undefined;
  adminLastName: string | undefined;
  adminEmail: string | undefined;
  adminPhone: string | undefined;
}

export interface EditOrgMutationResponse {
  data: {
    editPendingOrganization: string;
  };
}

const phoneNumberIsValidOrEmpty = (input: any) =>
  input === "" || phoneNumberIsValid(input);

export const pendingOrganizationSchema: yup.SchemaOf<PendingOrganizationFormValues> =
  yup.object().shape({
    name: yup.string().required("Organization name is required"),
    adminFirstName: yup.string(),
    adminLastName: yup.string(),
    adminEmail: yup.string().email("A valid email address is required"),
    adminPhone: yup
      .mixed()
      .test("", "A valid phone number is required", phoneNumberIsValidOrEmpty),
  });
