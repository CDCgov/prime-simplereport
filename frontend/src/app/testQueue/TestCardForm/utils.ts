import { SymptomDefinitionMap } from "../../../patientApp/timeOfTest/constants";

export const parseSymptoms = (
  symptomsJsonString: string | null | undefined,
  symptomSelectionMap: SymptomDefinitionMap[]
) => {
  const symptoms: Record<string, boolean> = {};

  if (symptomsJsonString) {
    const parsedSymptoms: { [key: string]: string | boolean } =
      JSON.parse(symptomsJsonString);

    symptomSelectionMap.forEach((opt) => {
      const val = opt?.value;
      if (typeof parsedSymptoms[val] === "string") {
        symptoms[val] = parsedSymptoms[val] === "true";
      } else {
        symptoms[val] = parsedSymptoms[val] as boolean;
      }
    });
  } else {
    symptomSelectionMap.forEach((opt) => {
      symptoms[opt?.value] = false;
    });
  }
  return symptoms;
};
