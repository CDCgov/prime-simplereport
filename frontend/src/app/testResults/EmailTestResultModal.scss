.email-test-result-modal-content {
  background: #fff;
  max-width: 30em;
  max-height: 90vh;
  padding: 28px 24px;

  .header {
    font-size: 22px;
    padding-bottom: 12px;
  }

  .body {
    font-size: 16px;
    padding-bottom: 20px;

    .text {
      color: #565c65;
      padding-bottom: 20px;
    }
  }
}
