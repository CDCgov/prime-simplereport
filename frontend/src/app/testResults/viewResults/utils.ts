import { Location } from "react-router-dom";

import { getParameterFromUrl } from "../../utils/url";

export function getFiltersFromUrl(url: Location) {
  const patientId = getParameterFromUrl("patientId", url);
  const startDate = getParameterFromUrl("startDate", url);
  const endDate = getParameterFromUrl("endDate", url);
  const role = getParameterFromUrl("role", url);
  const disease = getParameterFromUrl("disease", url);
  const result = getParameterFromUrl("result", url);
  const filterFacilityId = getParameterFromUrl("filterFacilityId", url);

  return {
    ...(patientId && { patientId: patientId }),
    ...(startDate && { startDate: startDate }),
    ...(endDate && { endDate: endDate }),
    ...(disease && { disease: disease }),
    ...(result && { result: result }),
    ...(role && { role: role }),
    ...(filterFacilityId && { filterFacilityId: filterFacilityId }),
  };
}
