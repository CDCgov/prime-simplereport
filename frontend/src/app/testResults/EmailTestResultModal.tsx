import Modal from "react-modal";

import Button from "../commonComponents/Button/Button";
import { formatFullName } from "../utils/user";
import "./TestResultCorrectionModal.scss";
import {
  useGetTestResultForResendingEmailsQuery,
  useResendTestResultsEmailMutation,
} from "../../generated/graphql";
import { showAlertNotification } from "../utils";
import "./EmailTestResultModal.scss";
import { formatDateLong } from "../utils/date";

interface Props {
  testResultId: string;
  closeModal: () => void;
}

export const EmailTestResultModal = ({ closeModal, testResultId }: Props) => {
  const [emailTestResult] = useResendTestResultsEmailMutation();

  const { data, loading } = useGetTestResultForResendingEmailsQuery({
    variables: { id: testResultId },
    fetchPolicy: "no-cache",
  });

  const patient = data?.testResult?.patient as Person;
  const dateTested = data?.testResult?.dateTested;

  return (
    <Modal
      isOpen={true}
      className="email-test-result-modal-content"
      overlayClassName="sr-test-correction-modal-overlay"
      contentLabel="Printable test result"
    >
      <div className="header">Email result?</div>
      {loading ? (
        <p>Loading</p>
      ) : (
        <>
          <div className="body">
            <div className="text">
              {formatFullName(patient)}'s test result from{" "}
              {formatDateLong(dateTested)} will be sent to the following emails:
            </div>
            {patient.emails?.map((email: string) => (
              <div key={email}>{email}</div>
            ))}
          </div>
          <div className="sr-test-correction-buttons">
            <Button variant="unstyled" label="Cancel" onClick={closeModal} />
            <Button
              label="Send result"
              onClick={() => {
                emailTestResult({
                  variables: { testEventId: testResultId },
                }).then((response) => {
                  const success =
                    response.data?.sendPatientLinkEmailByTestEventId;
                  showAlertNotification(
                    success ? "success" : "error",
                    success
                      ? "Emailed test results."
                      : "Failed to email test results."
                  );
                  window.scrollTo(0, 0);
                  closeModal();
                });
              }}
            />
          </div>
        </>
      )}
    </Modal>
  );
};

export default EmailTestResultModal;
