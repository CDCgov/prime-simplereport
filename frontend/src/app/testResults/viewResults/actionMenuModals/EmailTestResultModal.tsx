import React from "react";
import Modal from "react-modal";

import Button from "../../../commonComponents/Button/Button";
import { formatFullName } from "../../../utils/user";
import {
  GetTestResultForResendingEmailsDocument,
  useResendTestResultsEmailMutation,
} from "../../../../generated/graphql";
import { showAlertNotification } from "../../../utils/srToast";
import { formatDateLong } from "../../../utils/date";
import { QueryWrapper } from "../../../commonComponents/QueryWrapper";

import "./EmailTestResultModal.scss";

interface DetachedEmailTestResultModalProps {
  data: any;
  testResultId: string;
  closeModal: () => void;
}

const DetachedEmailTestResultModal = ({
  data,
  closeModal,
  testResultId,
}: DetachedEmailTestResultModalProps) => {
  const [emailTestResult] = useResendTestResultsEmailMutation();

  const patient = data?.testResult?.patient as Person;
  const dateTested = data?.testResult?.dateTested;

  return (
    <>
      <h1 className="font-sans-lg margin-top-0 margin-bottom-105 text-normal">
        Email result?
      </h1>
      <div className="body">
        <div className="text">
          {formatFullName(patient)}'s test result from{" "}
          {formatDateLong(dateTested)} will be sent to the following emails:
        </div>
        <ul className="usa-list usa-list--unstyled">
          {patient.emails?.map((email: string) => (
            <li className="line-height-sans-2" key={email}>
              {email}
            </li>
          ))}
        </ul>
      </div>
      <div className="sr-test-correction-buttons">
        <Button variant="unstyled" label="Cancel" onClick={closeModal} />
        <Button
          label="Send result"
          onClick={() => {
            emailTestResult({
              variables: { testEventId: testResultId },
            }).then((response) => {
              const success = response.data?.sendPatientLinkEmailByTestEventId;
              showAlertNotification(
                success ? "success" : "error",
                success
                  ? "Emailed test results."
                  : "Failed to email test results."
              );
              window.scrollTo(0, 0);
              closeModal();
            });
          }}
        />
      </div>
    </>
  );
};
export type EmailTestResultModalProps = Omit<
  DetachedEmailTestResultModalProps,
  "data"
> & {
  isOpen: boolean;
};

const EmailTestResultModal = (props: EmailTestResultModalProps) => (
  <Modal
    isOpen={props.isOpen}
    className="email-test-result-modal-content"
    overlayClassName="sr-test-correction-modal-overlay"
    contentLabel="Printable test result"
    onRequestClose={props.closeModal}
  >
    {props.testResultId && (
      <QueryWrapper<DetachedEmailTestResultModalProps>
        query={GetTestResultForResendingEmailsDocument}
        queryOptions={{ variables: { id: props.testResultId } }}
        Component={DetachedEmailTestResultModal}
        componentProps={{ ...props }}
      />
    )}
  </Modal>
);

export default EmailTestResultModal;
