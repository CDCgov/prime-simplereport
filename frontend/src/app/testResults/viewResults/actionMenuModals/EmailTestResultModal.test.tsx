import { render, screen, waitFor } from "@testing-library/react";
import { MockedProvider } from "@apollo/client/testing";

import * as srToast from "../../../utils/srToast";
import {
  GetTestResultForResendingEmailsDocument,
  ResendTestResultsEmailDocument,
} from "../../../../generated/graphql";
import { setup } from "../../../utils/jestHelpers";

import EmailTestResultModal, {
  EmailTestResultModalProps,
} from "./EmailTestResultModal";

const mockCloseModal = jest.fn();

jest.mock("react-modal", () => (props: any) => {
  return <div>{props.children}</div>;
});

window.scrollTo = jest.fn();

const getEmailTestResultModalWithMocks = (
  props: EmailTestResultModalProps,
  mocks: any = []
) => (
  <MockedProvider mocks={mocks}>
    <EmailTestResultModal {...props} />
  </MockedProvider>
);

const getTestResultForResendingEmailsMock = {
  request: {
    query: GetTestResultForResendingEmailsDocument,
    variables: { id: "super-fancy-id" },
  },
  result: {
    data: {
      testResult: {
        dateTested: "2021-11-02T14:39:15.472Z",
        patient: {
          firstName: "Zelda",
          middleName: "Francesca Holcomb",
          lastName: "Gordon",
          email: "gesezyx@mailinator.com",
          emails: ["gesezyx@mailinator.com"],
          __typename: "Patient",
        },
        __typename: "TestResult",
      },
    },
  },
};
const resendTestResultsEmailMock = {
  request: {
    query: ResendTestResultsEmailDocument,
    variables: { testEventId: "super-fancy-id" },
  },
  result: {
    data: { sendPatientLinkEmailByTestEventId: true },
  },
};

const resendTestResultsEmailFailureMock = {
  request: {
    query: ResendTestResultsEmailDocument,
    variables: { testEventId: "super-fancy-id" },
  },
  result: {
    data: { sendPatientLinkEmailByTestEventId: false },
  },
};

describe("EmailTestResultModal", () => {
  let alertSpy: jest.SpyInstance;

  beforeEach(() => {
    alertSpy = jest.spyOn(srToast, "showAlertNotification");
  });

  afterEach(() => {
    alertSpy.mockRestore();
  });

  it("should show render", async () => {
    const { container } = render(
      getEmailTestResultModalWithMocks(
        {
          isOpen: true,
          testResultId: "super-fancy-id",
          closeModal: mockCloseModal,
        },
        [getTestResultForResendingEmailsMock]
      )
    );

    await screen.findByText("Email result?");
    screen.getByText(
      "Zelda Francesca Holcomb Gordon's test result from November 2nd, 2021 will be sent to the following emails:"
    );
    screen.getByText("gesezyx@mailinator.com");
    screen.getByText("Send result");
    screen.getByText("Cancel");

    expect(container).toMatchSnapshot();
  });

  describe("clicking on Send result button", () => {
    it("should resend the test results email and show success message", async () => {
      const { user } = setup(
        getEmailTestResultModalWithMocks(
          {
            isOpen: true,
            testResultId: "super-fancy-id",
            closeModal: mockCloseModal,
          },
          [getTestResultForResendingEmailsMock, resendTestResultsEmailMock]
        )
      );

      await screen.findByText("Email result?");
      await user.click(screen.getByRole("button", { name: /send result/i }));
      await waitFor(() =>
        expect(alertSpy).toHaveBeenCalledWith(
          "success",
          "Emailed test results."
        )
      );
      expect(mockCloseModal).toHaveBeenCalled();
    });

    it("should show error message when failing to email", async () => {
      const { user } = setup(
        getEmailTestResultModalWithMocks(
          {
            isOpen: true,
            testResultId: "super-fancy-id",
            closeModal: mockCloseModal,
          },
          [
            getTestResultForResendingEmailsMock,
            resendTestResultsEmailFailureMock,
          ]
        )
      );

      await screen.findByText("Email result?");
      await user.click(screen.getByText("Send result"));
      await waitFor(() =>
        expect(alertSpy).toHaveBeenCalledWith(
          "error",
          "Failed to email test results."
        )
      );
      expect(mockCloseModal).toHaveBeenCalled();
    });
  });
});
