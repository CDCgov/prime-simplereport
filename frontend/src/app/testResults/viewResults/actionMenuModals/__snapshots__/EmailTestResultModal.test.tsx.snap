// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailTestResultModal should show render 1`] = `
<div>
  <div>
    <h1
      class="font-sans-lg margin-top-0 margin-bottom-105 text-normal"
    >
      Email result?
    </h1>
    <div
      class="body"
    >
      <div
        class="text"
      >
        Zelda Francesca Holcomb Gordon
        's test result from
         
        November 2nd, 2021
         will be sent to the following emails:
      </div>
      <ul
        class="usa-list usa-list--unstyled"
      >
        <li
          class="line-height-sans-2"
        >
          gesezyx@mailinator.com
        </li>
      </ul>
    </div>
    <div
      class="sr-test-correction-buttons"
    >
      <button
        class="usa-button usa-button--unstyled"
        type="button"
      >
        Cancel
      </button>
      <button
        class="usa-button"
        type="button"
      >
        Send result
      </button>
    </div>
  </div>
</div>
`;
